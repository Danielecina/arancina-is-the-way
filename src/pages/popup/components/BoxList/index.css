.box {
  color: white;
  line-height: 1;
  display: flex;
  flex-direction: column;
  gap: 16px;
  position: relative;

  padding: 16px;
  background: rgba(255, 255, 255, 0.2);
  backdrop-filter: blur(5px);
  border-radius: 16px;
}

.box .box-row {
  display: grid;
  grid-template-columns: 1fr;
  grid-auto-rows: auto auto;
  align-items: center;
  gap: 12px;
}

.box .box-row--title {
  font-size: 22px;
  font-weight: 900;
  display: flex;
  align-items: center;
  gap: 12px;
}

.box .box-row--actions {
  display: flex;
  align-items: center;
}
