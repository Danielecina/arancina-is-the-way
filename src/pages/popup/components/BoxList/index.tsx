import React, {ReactNode} from 'react'

import './index.css'

export interface RowType {
  id: string,
  content: ReactNode,
  actions?: ReactNode,
  extra?: ReactNode
}
export type BoxListType = {
  rows: Array<RowType>
}
const BoxList: React.FC<BoxListType> = ({rows}) => (
  <main className={'box'}>
    {rows.map(({content, actions, extra, id}) => (
      <BoxRow key={id}>
        <BoxRowContent>{content}</BoxRowContent>
        <BoxRowActions>{actions}</BoxRowActions>
        {extra}
      </BoxRow>
    ))}
  </main>
)

export default BoxList

export const BoxRow: React.FC = ({children}) => {
  return <div className={'box-row'}>{children}</div>
}

export const BoxRowContent: React.FC = ({children}) => {
  return <div className={'box-row--title'}>{children}</div>
}

export const BoxRowActions: React.FC = ({children}) => {
  return <div className={'box-row--actions'}>{children}</div>
}
