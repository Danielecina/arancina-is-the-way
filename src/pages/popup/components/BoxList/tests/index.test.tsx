import React from 'react'
import renderer from 'react-test-renderer'

import BoxList, {BoxRow, BoxRowActions, BoxRowContent} from '..'

const rows = [
  {
    id: 'test',
    content: 'test-content',
    actions: 'test-actions',
    extra: 'test-extra'
  }
]

describe('BoxList component', () => {
  test('snapshot', () => {
    const element = renderer.create(<BoxList rows={rows} />).toJSON()
    expect(element).toMatchSnapshot()
  })

  test('boxRow snapshot', () => {
    const element = renderer.create(<BoxRow>{'BoxRow children'}</BoxRow>).toJSON()
    expect(element).toMatchSnapshot()
  })

  test('BoxRowActions snapshot', () => {
    const element = renderer.create(<BoxRowActions>{'BoxRowActions children'}</BoxRowActions>).toJSON()
    expect(element).toMatchSnapshot()
  })

  test('BoxRowActions snapshot', () => {
    const element = renderer.create(<BoxRowContent>{'BoxRowContent children'}</BoxRowContent>).toJSON()
    expect(element).toMatchSnapshot()
  })
})
