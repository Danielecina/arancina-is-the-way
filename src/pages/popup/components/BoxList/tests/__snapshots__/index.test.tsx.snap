// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxList component BoxRowActions snapshot 1`] = `
<div
  className="box-row--actions"
>
  BoxRowActions children
</div>
`;

exports[`BoxList component BoxRowActions snapshot 2`] = `
<div
  className="box-row--title"
>
  BoxRowContent children
</div>
`;

exports[`BoxList component boxRow snapshot 1`] = `
<div
  className="box-row"
>
  BoxRow children
</div>
`;

exports[`BoxList component snapshot 1`] = `
<main
  className="box"
>
  <div
    className="box-row"
  >
    <div
      className="box-row--title"
    >
      test-content
    </div>
    <div
      className="box-row--actions"
    >
      test-actions
    </div>
    test-extra
  </div>
</main>
`;
